### ISSUE TITLE
@Modifiable equalTo

### ISSUE THREAD

**Issue by pushbit** [immutables/immutables#322]
The implementation of equalTo doesn't look quite right.

```java
	@Modifiable
	abstract class Bar {
		abstract List<Integer> bar();
	}
```
creates
```java
  private boolean equalTo(ModifiableBar another) {
    List<Integer> 
  bar = new ArrayList<Integer>();
    return bar.equals(another.bar);
  }
```
I think the `bar` local variable shouldn't be there and the comparisons of `bar` also need a `null` check.

**Comment by elucash** [comment]
yes, this part require an adequate check for null

**Comment by pushbit** [comment]
The updated check is better, but when both `bar` fields are null this will return false.  I think either `another.bar` needs to also be changed to `emptyList()` when it's `null` or there needs to be a check for when both fields are `null`.

```java
    List<Integer> bar = this.bar != null
        ? this.bar
        : Collections.<Integer>emptyList();
    return bar.equals(another.bar);
```

**Comment by elucash** [comment]
released in 2.2 Thank you!
